File: BleModule/src/main/java/com/mpdc4gsr/ble/callback/ScanListener.java
package com.mpdc4gsr.ble.callback;

import android.Manifest;

import com.mpdc4gsr.ble.Device;

public interface ScanListener {

    int ERROR_LACK_LOCATION_PERMISSION = 0;

    int ERROR_LOCATION_SERVICE_CLOSED = 1;

    int ERROR_SCAN_FAILED = 2;

    void onScanStart();

    void onScanStop();

    @Deprecated
    default void onScanResult(Device device) {
    }

    void onScanResult(Device device, boolean isConnectedBySys);

    void onScanError(int errorCode, String errorMsg);
}

File: BleModule/src/main/java/com/mpdc4gsr/ble/callback/ScanListener.kt
package com.mpdc4gsr.ble.callback

import com.mpdc4gsr.ble.Device

interface ScanListener {
    fun onScanStart()

    fun onScanStop()

    @Deprecated("")
    fun onScanResult(device: Device?) {
    }

    fun onScanResult(device: Device?, isConnectedBySys: Boolean)

    fun onScanError(errorCode: Int, errorMsg: String?)

    companion object {
        const val ERROR_LACK_LOCATION_PERMISSION: Int = 0

        const val ERROR_LOCATION_SERVICE_CLOSED: Int = 1

        const val ERROR_SCAN_FAILED: Int = 2

        const val ERROR_LACK_BLUETOOTH_PERMISSION: Int = 3
    }
}
